scripts/wrappers/join.py

```diff
@@ -760,6 +761,14 @@ def join_dqlite(connection_parts, verify=False, worker=False):
         worker=worker,
     )
 
+    # Get the cluster_cidr from kube-proxy args
+    cluster_cidr = get_cluster_cidr()
+
+    if "cluster_cidr" in info and info["cluster_cidr"] != cluster_cidr:
+        print("Joining cluster failed. CIDR for the nodes does not match.")
+        print(f"Cluster CIDR: {info['cluster_cidr']} -- Node CIDR:  {cluster_cidr}")
+        exit(4)
+
```

Comment:
Is it not enough to just update the cluster-cidr in the local node kube-proxy?

Reply:
+1 lets try to update the cidr of the joining node with the one we get from the cluster we are joining.